plugins/by-name/haskell-tools/default.nix

```
@@ -0,0 +1,64 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+lib.nixvim.plugins.mkNeovimPlugin rec {
+  name = "haskell-tools";
```

Can we avoid the `rec` if possible? They look neat and elegant at first, until you try to read existing code that uses `with` or `rec` (without an LSP).

I don't mind them when applied to a small scope/value, but across an entire file they aren't ideal.

```suggestion
let
  name = "haskell-tools";
in
lib.nixvim.plugins.mkNeovimPlugin {
  inherit name;
```

You could use `lib.fix` here, but that feels over-engineered for this case:

```nix
lib.nixvim.plugins.mkNeovimPlugin (lib.fix (self: {
```

---

No problem. I like recursion but don't mind changing it.